sotopia/benchmark/cli.py

```diff
@@ -0,0 +1,302 @@
+                        tag=tag,
+                    )
+                )
+                env_agent_combo_batch = []
+        else:
+            if env_agent_combo_batch:
+                logging.info(
+                    f"Running batch of {batch_size} episodes: {env_agent_combo_batch}"
+                )
+                loop.run_until_complete(
+                    run_async_server(
+                        model_dict=model_names,
+                        sampler=BaseSampler[Observation, AgentAction](),
+                        env_agent_combo_list=env_agent_combo_batch,
+                        push_to_db=push_to_db,
+                        tag=tag,
+                    )
+                )
+            # remove episodes that has bad rewards
+            simulated_episodes = EpisodeLog.find(EpisodeLog.tag == tag).all()
+            valid_episodes = [
+                not isinstance(relevant_episode.rewards[0], float)  # type: ignore
+                for relevant_episode in simulated_episodes
+            ]
+            for valid, episode in zip(valid_episodes, simulated_episodes):
+                if not valid:
+                    pk = episode.pk
+                    assert isinstance(pk, str)
+                    EpisodeLog.delete(pk)
+
+            env_agent_combo_list = _list_all_env_agent_combo_not_in_db(
+                model_names=model_names,
+                tag=tag,
+                env_agent_combo_storage_list=benchmark_combo,
+            )
+            env_agent_combo_batch = []
+            number_of_fix_turns += 1
+            if len(env_agent_combo_list) == 0 or number_of_fix_turns >= 5:
+                rewards_dict = get_avg_reward(
+                    simulated_episodes,  # type: ignore
+                    model_names["test_model"],
+                )
+                rewards_dict["model_name"] = model_names["test_model"]  # type: ignore
+                rewards_dict["episode_count"] = len(simulated_episodes)
+                rich.print(rewards_dict)
+                return
+
+
+def _set_up_logs(
+    *,
+    log_file_level: int = logging.DEBUG,
+    log_rich_level: int = logging.INFO,
+    log_format: str = "%(asctime)s - %(levelname)s - %(name)s - %(message)s",
+    log_file: str = datetime.now().strftime("./logs/%H_%M_%d_%m_%Y.log"),
+    print_logs: bool = False,
+) -> None:
+    # date and message only
+    logging_path = Path(log_file)
+    logging_path.parent.mkdir(parents=True, exist_ok=True)
+
+    logging.basicConfig(
+        level=log_file_level,
+        format=log_format,
+        datefmt="[%X]",
+        handlers=[
+            FileHandler(logging_path),
+            RichHandler(level=log_rich_level) if print_logs else RichHandler(level=100),
+        ],
+    )
+
+
+@app.command()
+def cli(
+    model: str = typer.Option(..., help="The language model you want to benchmark."),
+    partner_model: str = typer.Option(
+        "meta-llama/Llama-3-70b-chat-hf",
```

Comment:
Why the provider's name is dropped here?

Reply:
check together ai's doc

Reply:
But it won't get into this branch in generate.py:  `if "together_ai" in model_name:`

Reply:
oh right, damn

Reply:
So why it works on your machine? The current generate.py should have already dealt with this by removing the first segment.

Reply:
that's a good q, why?

Reply:
fixed